QualityControl/lib/api.js

```diff
@@ -149,13 +149,10 @@ function errorHandler(err, res, status = 500) {
  */
 function getTagsFromServices(services) {
   const prefix = model.queryPrefix;
-  const tags = [];
-  for (const serviceName in services) {
-    if (services[serviceName] && services[serviceName].Tags && services[serviceName].Tags.length > 0) {
-      const tagsToBeAdded = services[serviceName].Tags;
-      tagsToBeAdded.filter((tag) => tag.startsWith(prefix))
-        .forEach((tag) => tags.push({name: tag}));
-    }
-  }
+  const tags = Object.entries(services)
+    .filter(([key]) => key.startsWith(prefix))
+    .map(([, value]) => value)
+    .flat()
+    .reduce((accu, tag) => accu.concat({name: tag}), []);
```

Comment:
Rather than using an accumulator to concatenate I think it is better to use `map`
`  .map((tag) => ({name: tag}))`

Reply:
Yes, indeed, it's more elegant